### PR TITLE
Add pagination support to index

### PR DESCRIPTION
**Pull request by felipefreitag**

### CODE REVIEW

File: todos/app/resources/projects/resource.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 module Projects
   class Resource < ApplicationResource
+    filter_by :page, optional: true
```
Comment (danielweinmann) [inline todos/app/resources/projects/resource.rb:5]: I think this should be a different method called `paginate`, where you set the (optional) options for Kaminari. I also think the default should be to paginate: you would only use the `paginate` method if you want to customize Kaminari. What do you think?

**Review by danielweinmann — CHANGES_REQUESTED** [review]

File: todos/app/resources/projects/resource.rb (hunk 1 above)
Comment (felipefreitag) [reply todos/app/resources/projects/resource.rb:5]: Agreed, I just didn't find where to set this default :page attribute. Inside Filters?

Comment (felipefreitag) [reply todos/app/resources/projects/resource.rb:5]: We also need to accept a per_page param.

Comment (felipefreitag) [reply todos/app/resources/projects/resource.rb:5]: This was a hack to have the easiest way to permit that param on json schema.

File: spec/todos/projects/index_spec.rb [hunk 2]
```diff
@@ -91,4 +91,63 @@
...
+
+    context 'when first page' do
+      before do
+        get '/projects?page=1'
+      end
+
+      it { is_expected.to have_http_status(:ok) }
+      it { expect(JSON.parse(response.body).length).to eq(25) }
+      it { expect(response.headers['Total']).to eq('30') }
+      it { expect(response.headers['Per-Page']).to eq('25') }
```
Comment (AlanMaik) [inline spec/todos/projects/index_spec.rb:120]: I think there was a test here for the links, what do you think about adding it?
an example of what this test should look like:
```ruby it 'expected links' do
      expect(response['Link'])
        .to eq(
          '<http://www.example.com/diamonds?page=2>; rel="last", ' \
          '<http://www.example.com/diamonds?page=2>; rel="next"'
        )
    end

Comment (felipefreitag) [reply spec/todos/projects/index_spec.rb:120]: Aren't those links only for Rails views that we don't use?

File: lib/croods/controller/collection.rb [hunk 3]
```diff
@@ -5,10 +5,22 @@ class Controller < ActionController::API
     module Collection
       protected
 
+      def page_param
```
Comment (felipefreitag) [inline lib/croods/controller/collection.rb:8]: @danielweinmann Should these 2 methods live here or inside Resource::Pagination?

Comment (danielweinmann) [reply lib/croods/controller/collection.rb:8]: I like them here

**Review by danielweinmann — APPROVED** [review]

**Pull request merged**